### PR TITLE
Add core/list block to the email editor

### PR DESCRIPTION
**Pull request by lysyjan**
## Description

_N/A_

## Code review notes

_N/A_

## QA notes

_N/A_

## Linked PRs

_N/A_

## Linked tickets

_N/A_

## After-merge notes

_N/A_

## Tasks

- [x] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] I added sufficient test coverage
- [x] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

### CODE REVIEW

File: mailpoet/lib/EmailEditor/Integrations/Core/Renderer/Blocks/ListBlock.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+use MailPoet\EmailEditor\Engine\SettingsController;
+
+// We have to avoid using keyword `List`
+class ListBlock implements BlockRenderer {
+  public function render($blockContent, array $parsedBlock, SettingsController $settingsController): string {
+    $contentStyles = $settingsController->getEmailContentStyles();
+    return str_replace('{list_content}', $blockContent, $this->prepareColumnTemplate($parsedBlock, $contentStyles));
+  }
+
+  private function prepareColumnTemplate(array $parsedBlock, array $contentStyles): string {
```
Comment (costasovo) [inline mailpoet/lib/EmailEditor/Integrations/Core/Renderer/Blocks/ListBlock.php]: I think we should come up with a better name, `prepareColumnTemplate` maybe something like `getBlockWrapper` or `getStyleResetWrapper` or something else. What do you think?

File: mailpoet/assets/js/src/email-editor/engine/components/block-editor/index.scss [hunk 2]
```diff
@@ -54,6 +54,18 @@
...
+    margin: 0;
+    padding: 0;
+  }
+
+  ul,
+  ol,
+  ul.has-background,
+  ol.has-background {
+    padding-left: 40px;
+  }
```
Comment (costasovo) [inline mailpoet/assets/js/src/email-editor/engine/components/block-editor/index.scss]: Could you please add comments for both these rules?

File: mailpoet/lib/EmailEditor/Integrations/Core/Renderer/Blocks/ListBlock.php [hunk 3]
```diff
@@ -26,9 +26,19 @@ private function prepareColumnTemplate(array $parsedBlock, array $contentStyles)
...
+        cellspacing="0"
+        role="presentation"
+        style="' . $this->convertStylesToString($styles) . '"
+      >
+        <tr>
+          <td>
+            {list_content}
+          </td>
+        </tr>
+      </table>
```
Comment (costasovo) [inline mailpoet/lib/EmailEditor/Integrations/Core/Renderer/Blocks/ListBlock.php:41]: I see !fixup in the commit message, so I guess you need to rebase

File: mailpoet/lib/EmailEditor/Engine/Renderer/styles.css [hunk 4]
```diff
@@ -33,7 +33,16 @@ img {
...
+  margin: 0;
+}
+
+h1,
+h2,
+h3,
+h4,
+h5,
+h6 {
+  margin: 0;
```
Comment (costasovo) [inline mailpoet/lib/EmailEditor/Engine/Renderer/styles.css]: Could you please comment why we reset this? 🙏

File: mailpoet/assets/js/src/email-editor/engine/components/block-editor/block-editor.tsx [hunk 5]
```diff
@@ -90,6 +90,7 @@ export function BlockEditor() {
       width: layoutStyles.width,
       display: 'flex',
       flexFlow: 'column',
+      'box-sizing': 'border-box',
```
Comment (costasovo) [inline mailpoet/assets/js/src/email-editor/engine/components/block-editor/block-editor.tsx]: Could you please comment what it solves?

**Review by costasovo — COMMENTED** [review]
Thanks for adding the support for the list!

**Comment by lysyjan** [comment]
@costasovo I hope I have addressed all your remarks. Thanks for a great review 👍

**Review by costasovo — APPROVED** [review]
Good job!

**Pull request merged**